core/src/main/java/org/apache/iceberg/stats/BlobMetadata.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+
+public class BlobMetadata {
+  private final String type;
+  private final Set<Integer> columns;
+  private final long offset;
+  private final long length;
+  private final String compressionCodec;
+
+  public BlobMetadata(String type, Set<Integer> columns, long offset, long length, @Nullable String compressionCodec) {
```

Comment:
None is not a compression. The file format spec defines there should be no mention of compression coded, when blob data is not compressed. Internally, in Java code we could still have an enum with additional value, but it would need to require special casing when reading, writing and compressing. Originally i had `Optional` here, but got a comment to use a nullable value instead.

Reply:
I agree with Kyle. I think this would be cleaner if you added `NONE` and always passed the codec as the enum.

Reply:
Be your way